osrf/vrx
# Update Follow-the-path scoring

Fixes #623 .

This patch updates the Follow-the-path task according to the discussion in issue #623 .

### How to test it?

Launch the teleoperation node:
```
ros2 launch vrx_gz usv_joy_teleop.py
```

Launch a practice world. E.g.:
```
ros2 launch vrx_gz competition.launch.py world:=practice_2022_follow_path2_task
```

Verify the following things:

* Make sure that if you skip the first gate, nothing happens (no gates crossed or invalid reported on the console).
* Cross the starting gate (`+10` points).
* Make sure that if you cross another gate you get `10` points + a bonus. The bonus is declared as a parameter in the SDF.
* Make sure that if you skip a gate, you'll get 10 points for crossing the next gate and no bonus.
* Make sure that if you invalidate a gate (crossing it backwards), it no longer produces points even if you later cross it properly.
* Make sure that when you cross the last gate, the task terminates.
* Make sure that if you hit an obstacle, you get a penalty of `-3` points.

Some notes:

* I bumped the teleoperation speed a bit.
* I left a few `gzdbg` in the plugin to help us debug the bonuses, penalties, etc.
